compiler/x/codegen/OMRMachine.cpp

```diff
@@ -859,7 +865,9 @@ TR::RealRegister *OMR::X86::Machine::freeBestGPRegister(TR::Instruction
             }
          else
             {
-            location = self()->cg()->allocateSpill(16, false, &offset);
+            int32_t size = self()->cg()->comp()->target().cpu.supportsAVX() ? 32 : 16;
+            size = self()->cg()->comp()->target().cpu.supportsFeature(OMR_FEATURE_X86_AVX512F) ? 64 : size;
+            location = self()->cg()->allocateSpill(size, false, &offset);
```

Comment:
Does this mean that even for registers used for 128-bit vectors we will spill them as 512-bit vectors on the stack on hardware that supports AVX512? (since the check is on whether AVX/AVX512 is supported, not the size of the register it is assigned to).  If so, this is less than ideal and we should try and refine that.